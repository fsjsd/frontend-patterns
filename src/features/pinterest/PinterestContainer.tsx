import React, { Suspense } from 'react'
import { ContentWrapper } from '../../ux/ContentWrapper'
import Loading from '../../ux/Loading'
const Pinterest = React.lazy(() => import('./Pinterest'))

const PinterestContainer = () => {
  return (
    <ContentWrapper
      title='Pinterest'
      codeLink="/features/pinterest"
      markDownPromise={import('./requirements.md')}
    >
      <Suspense fallback={<Loading />}>
        <div role="main">
          <Pinterest />
        </div>
      </Suspense>
    </ContentWrapper>
  )
}

export default PinterestContainer