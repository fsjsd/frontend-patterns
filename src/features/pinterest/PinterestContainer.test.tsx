import React from 'react'
import PinterestContainer from "./PinterestContainer";
import { render, waitFor } from '@testing-library/react'

describe("Pinterest", () => {
  it("renders and matches snapshot", async () => {
    const { container, queryByRole } = render(<PinterestContainer />)
    await waitFor(() => expect(queryByRole("main")).toBeInTheDocument());
    expect(container).toMatchSnapshot();

  })
})