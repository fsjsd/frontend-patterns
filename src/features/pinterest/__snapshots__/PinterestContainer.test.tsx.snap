// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pinterest renders and matches snapshot 1`] = `
<div>
  <header
    class="sc-bczRLJ dmSzxI"
  >
    <div
      class="sc-gsnTZi hCGKc"
      role="heading"
    >
      Pinterest
    </div>
    <div
      class="sc-dkzDqf gedmVz"
    >
      <div
        class="sc-hKMtZM dNwpDv"
        role="button"
        title="Run demo"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M10 16.5l6-4.5-6-4.5zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
          />
        </svg>
      </div>
      <div
        class="sc-hKMtZM itnQjY"
        role="button"
        title="View notes"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M3 18h12v-2H3v2zM3 6v2h18V6H3zm0 7h18v-2H3v2z"
          />
        </svg>
      </div>
      <div
        class="sc-hKMtZM itnQjY"
        role="button"
        title="View code on github"
      >
        <a
          href="https://github.com/fsjsd/frontend-patterns/tree/main/src/features/pinterest"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M9.4 16.6L4.8 12l4.6-4.6L8 6l-6 6 6 6 1.4-1.4zm5.2 0l4.6-4.6-4.6-4.6L16 6l6 6-6 6-1.4-1.4z"
            />
          </svg>
        </a>
      </div>
    </div>
  </header>
  <section
    class="sc-jSMfEi foNrii"
    role="document"
  >
    <div
      role="main"
      style=""
    >
      <div
        class="sc-iBkjds fPMrdA"
      >
        Coming soon
      </div>
    </div>
  </section>
</div>
`;
